### PR TITLE
Add Exception

### PR DESCRIPTION
**Pull request by Resurr3ction**

### CODE REVIEW

**Comment by Resurr3ction** [comment]
Resolves #75

File: projects/ACore/include/ACore/Exception.hpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace acore
+{
+//! The Exception is a standard layout streamable
+//! exception.
+//!
+//! Unlike the <a href="http://en.cppreference.com/w/cpp/error/exception">std::exception</a>
+//! the Exception is a standard layout class and
+//! can be safely used (and thrown) across library
+//! boundaries. Similar to <a href="http://en.cppreference.com/w/cpp/error/exception">std::logic_error</a>
```
Comment (Agnesoft) [inline projects/ACore/include/ACore/Exception.hpp:17]: Similar to `std::exception`

File: projects/ACore/include/ACore/Exception.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+class Exception
+{
+public:
+    //! Returns the null terminated message.
+    [[nodiscard]] auto what() const noexcept -> const char *
+    {
+        return mMessage.c_str();
+    }
+
+    //!Appends the arbitrary \a text to the exception's
```
Comment (Agnesoft) [inline projects/ACore/include/ACore/Exception.hpp:42]: Space between comment and `Appends`

File: projects/ACore/test/ExceptionTest.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#include "pch.hpp"
+
+#include <Catch2/catch.hpp>
+
+namespace exceptiontest
+{
+TEST_CASE("[acore::Exception]")
+{
+#ifndef _WIN32
```
Comment (Agnesoft) [inline projects/ACore/test/ExceptionTest.cpp:9]: MSVC `std::string` is standard layour now, the `ifdef` should no longer be needed.

**Pull request merged**